### ISSUE TITLE
[BUG] v1.4.x setting disable-snapshot-purge is not supported error

### ISSUE THREAD

**Issue by ChanYiLin** [longhorn/longhorn#7828]
## Describe the bug

Install longhorn with `v1.4.x-head` images got errors and longhorn-managers failed to start
```
I0201 09:19:26.365786       1 shared_informer.go:247] Caches are synced for longhorn-system-restore 
time="2024-02-01T09:19:26Z" level=fatal msg="Error starting manager: admission webhook \"validator.longhorn.io\" denied the request: failed to set the setting disable-snapshot-purge with invalid value false: value false of settings disable-snapshot-purge is invalid: setting disable-snapshot-purge is not supported" func=main.main.DaemonCmd.func3 file="daemon.go:79"
```

## To Reproduce

Install longhorn with `v1.4.x-head`

## Expected behavior

should success to install

**Comment by ChanYiLin** [comment]
cc @ejweber

**Comment by ejweber** [comment]
I tried to recreate this but was (mostly) unsuccessful.

1. `git checkout v1.4.x`
2. Replace all instances of `longhornio/longhorn-manager:v1.4.4` with `longhornio/longhorn-manager:v1.4.x-head` in `deploy/longhorn.yaml`. (There are 10 by my IDE's count.)
3. `kubectl apply -f deploy/longhorn.yaml`
4. Check behavior with the following commands:

```
eweber@laptop:~/longhorn> kl get pod -l app=longhorn-manager
NAME                     READY   STATUS    RESTARTS        AGE
longhorn-manager-48xx5   1/1     Running   0               6m5s
longhorn-manager-6w9bn   1/1     Running   1 (5m45s ago)   6m5s
longhorn-manager-7jdqr   1/1     Running   1 (5m45s ago)   6m5s
eweber@laptop:~/longhorn> kl get setting | grep disable-snapshot-purge
disable-snapshot-purge                                            false                                         5m50s
```

I DID manage to reproduce the issue when I only changed the image of the `longhorn-manager` pod to `longhornio/longhorn-manager:v1.4.x-head`, but not the webhook pods. Is it possible that this was the cause when you ran into the error?

I also noticed that I could hit the issue temporarily when redeploying with `longhornio/longhorn-manager:v1.4.x-head` after having previously deployed with `longhornio/longhorn-manager:v1.4.4`. It seems that the new `longhorn-manager` pod may be able to briefly use the old webhook during the upgrade. I think I remember a previous issue in which we discussed this, but I couldn't locate it quickly. I can investigate this further if you think its relevant.
